hudi-flink/src/main/java/org/apache/hudi/table/format/cow/ParquetSplitReaderUtil.java

```diff
@@ -104,7 +106,7 @@ public static ParquetColumnarRowSplitReader genPartColumnarRowReader(
         .filter(nonPartNames::contains).collect(Collectors.toList());
 
     int[] selParquetFields = selNonPartNames.stream()
-        .mapToInt(nonPartNames::indexOf)
+        .mapToInt(fullNames::indexOf)
         .toArray();
```

Comment:
indexOf is not very efficient

Reply:
Grazie! I got it. If there are too many columns, use the indexOf or contains method to judge, the time complexity is O (n^2).